pages/api/auth/login.ts

```diff
@@ -78,3 +70,22 @@ export default async function handleR(
     return handleError(res, e);
   }
 }
+
+async function logUserIn(existingSession: Session | null, userUUID: string) {
+  // If the existing session was found update it to be logged in
+  if (existingSession) {
+    await prisma.session.update({
+      where: { id: existingSession.id },
+      data: { isLoggedIn: true },
+    });
+    return existingSession.id;
+  }
+
+  // Create a new session if the existing session was not found
+  const newSession = await luciaShortSession.createSession(userUUID, {
+    userUUID: userUUID,
+    isLoggedIn: true,
+  });
+
+  return newSession.id;
+}
```

Comment:
Toteutin ehdotuksesi tällä tavalla.

Ehdotus: 

```ts
async function logUserIn(
  lucia: typeof luciaShortSession,
  existingSession: string | null,
  userUUID: string,
) {
  let sessionId: string | undefined;

  // Use existing session if it's defined and found from the database
  if (existingSession)
    sessionId = (
      await prisma.session.findUnique({ where: { id: existingSession } })
    )?.id;

  // If the existing session was found update it to be logged in
  if (sessionId)
    await prisma.session.update({
      where: { id: sessionId },
      data: { isLoggedIn: true },
    });
  // Create a new session if the existing session was not found
  else
    sessionId = (
      await lucia.createSession(userUUID, {
        userUUID: userUUID,
        isLoggedIn: true,
      })
    ).id;

  return sessionId;
}
```

Oliko `lucia`n ottaminen parametrina sen takia, että käytetään samaa instancea Lucia-classista, kuin mitä itse `loginHandler`-funktiossa on? Vai voisiko sen ottaa suoraan importista (`luciaShortSession`)? 

Tein uuden typen `Session`, joka sisältää kaiken tarvittavan tiedon Sessionista. Vaihdoin `existingSession`in käyttämään `Session`-typeä mielummin kuin `stringiä`. Sillä `existingSession`han sisältää kaiken saman tiedon, kuin jos haettaisiin ne uudestaan

Muuten on samanlainen kuin ehdottamasi

Reply:
> Oliko `lucia`n ottaminen parametrina sen takia, että käytetään samaa instancea Lucia-classista, kuin mitä itse `loginHandler`-funktiossa on? Vai voisiko sen ottaa suoraan importista (`luciaShortSession`)?

`lucia` otettiin parametrina sen takia, etten tiennyt pitäisikö käyttää `luciaShortSession` vai `luciaLongSession`. Nyt "Muista minut" ei taida toimi oiein, kun sessio luodaan aina `luciaShortSession` kautta. `lucia`n sijaan parametriin voi laittaa myös `rememberMe` arvon.

> Tein uuden typen `Session`, joka sisältää kaiken tarvittavan tiedon Sessionista. Vaihdoin `existingSession`in käyttämään `Session`-typeä mielummin kuin `stringiä`. Sillä `existingSession`han sisältää kaiken saman tiedon, kuin jos ne haettaisiin uudestaan tietokannasta

Tämä oli erinomainen muutos 👍

Reply:
> `lucia` otettiin parametrina sen takia, etten tiennyt pitäisikö käyttää `luciaShortSession` vai `luciaLongSession`. Nyt "Muista minut" ei taida toimi oiein, kun sessio luodaan aina `luciaShortSession` kautta. `lucia`n sijaan parametriin voi laittaa myös `rememberMe` arvon.

Totta, ei muuten toimikkaan! Hyvä löytö! 

Laitoin `rememberMe`-parametrin ja tein samalla tavalla kuin `loginHandler`-funktiossa. Kommentoin tämän kun pushaan :+1: